tools/setup/emoji/emoji_setup_utils.py

```diff
@@ -69,10 +71,11 @@ def get_emoji_code(emoji_dict: Dict[str, Any]) -> str:
     # should have a colorful display. Hence we can continue using emoji characters
     # without appending emoji presentation selector.
     # (http://unicode.org/reports/tr51/index.html#Presentation_Style)
-    # If `non_qualified` field is present and not None return it otherwise
-    # return `unified` field.
-    emoji_code = emoji_dict.get("non_qualified") or emoji_dict["unified"]
-    return emoji_code.lower()
+    # There is a `non_qualified` field on `emoji_dict` but it's inconsistently
+    # present, so we'll always use the unqualified emoji by unqualifying it
+    # ourselves. This gives us more consistent behaviour between emojis, and doesn't
+    # rely on the incomplete upstream package.
+    return emoji_to_hex_codepoint(unqualify_emoji(hex_codepoint_to_emoji(emoji_dict["unified"])))
```

Comment:
This comment might be a good place to write out a bit more what the difference between `non_qualifed` and `unified` is, or link to it.

Reply:
It's directly related to the `emoji presentation selector` which is discussed directly above with a link, but I can make the connection more clear with a bit more comment text.